src/main/scala/org/broadinstitute/dsde/workbench/sam/util/DatabaseSupport.scala

```diff
@@ -11,9 +12,12 @@ trait DatabaseSupport {
   protected val cs: ContextShift[IO]
   protected val dbRef: DbReference
 
-  protected def runInTransaction[A](databaseFunction: DBSession => A): IO[A] = {
-    cs.evalOn(ecForDatabaseIO)(IO {
-      dbRef.inLocalTransaction(databaseFunction)
-    })
+  protected def runInTransaction[A](dbQueryName: String, samRequestContext: SamRequestContext)(databaseFunction: DBSession => A): IO[A] = {
+    val spanName = "postgres-" + dbQueryName
+    traceIOWithContext(spanName, samRequestContext) { _ =>
```

Comment:
I think this should be after the `cs.evalOn` call. The way it currently is, the span includes time to wait for the context switch.